packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -374,6 +375,20 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-POST-V1-09
+    LROPostFinalStateViaProperty: {
+      description:
+        "A LRO POST MUST have long-running-operation-options specified and MUST have location header in the final-state-via property.",
+      message: "{{error}}",
+      severity: "error",
+      resolved: false,
+      formats: [oas2],
+      given: ["$[paths,'x-ms-paths'].*[post][?(@property === 'x-ms-long-running-operation' && @ === true)]^"],
```

Comment:
See if it still works when you remove the carat (^) and the filter. The carat seems to cancel out the filter
```suggestion
      given: ["$[paths,'x-ms-paths'].*[post]"],
```